src/HTTPure/Body.purs

```diff
@@ -1,25 +1,64 @@
 module HTTPure.Body
   ( Body
+  , string
+  , binary
+  , isString
+  , isBinary
+  , fromString
+  , fromBinary
   , read
   , write
+  , size
   ) where
 
 import Prelude
 
 import Data.Either as Either
+import Data.String as String
 import Effect as Effect
 import Effect.Aff as Aff
 import Effect.Ref as Ref
+import Node.Buffer as Buffer
 import Node.Encoding as Encoding
 import Node.HTTP as HTTP
 import Node.Stream as Stream
 
 -- | The `Body` type is just sugar for a `String`, that will be sent or received
 -- | in the HTTP body.
-type Body = String
+data Body
+  = StringBody String
+  | BinaryBody Buffer.Buffer
+
+-- | Construct a `Body` from a string
+string :: String -> Body
+string = StringBody
+
+-- | Construct a `Body` from a `Buffer`
+binary :: Buffer.Buffer -> Body
+binary = BinaryBody
+
+-- | Returns `true` when the `Body` is a string
+isString :: Body -> Boolean
+isString (StringBody _) = true
+isString _ = false
+
+-- | Returns `true` when the `Body` is binary
+isBinary :: Body -> Boolean
+isBinary (BinaryBody _) = true
+isBinary _ = false
```

Comment:
I don't think these boolean functions should be part of the API; folks should be using pattern matching for this.  So for instance (from below), instead of:

```purs
bodyShouldEqual :: Body.Body -> String -> Aff.Aff Unit
￼ bodyShouldEqual body str =
￼   if Body.isString body
￼   then Unsafe.unsafePartial Body.fromString body ?= str
￼   else do ...
```

it should be:

```purs
bodyShouldEqual :: Body.Body -> String -> Aff.Aff Unit
bodyShouldEqual (StringBody body) str = body ?= str
bodyShouldEqual (BinaryBody body) str = do ...
```

Similarly, instead of 

```purs
  Spec.it "has the right body" do
￼      resp <- Response.response 123 "test"
￼     Body.isString resp.body ?= true
￼     bodyShouldEqual resp.body "test"
```

We should have something like:

```purs
Spec.it "has the right body" do
    resp <- Response.response 123 "test"
    case resp.body of
        (String body) -> bodyShouldEqual body "test"
        _ -> false
```